packages/sdk/src/protocols/base_protocol.ts

```diff
@@ -1,15 +1,211 @@
+import type { Peer, PeerId } from "@libp2p/interface";
+import { ConnectionManager } from "@waku/core";
+import { BaseProtocol } from "@waku/core/lib/base_protocol";
 import { IBaseProtocolSDK } from "@waku/interfaces";
+import { Logger } from "@waku/utils";
 
 interface Options {
   numPeersToUse?: number;
+  maintainPeersInterval?: number;
 }
 
 const DEFAULT_NUM_PEERS_TO_USE = 3;
+const DEFAULT_MAINTAIN_PEERS_INTERVAL = 30_000;
 
 export class BaseProtocolSDK implements IBaseProtocolSDK {
-  public readonly numPeers: number;
+  public readonly numPeersToUse: number;
+  private peers: Peer[] = [];
+  private maintainPeersIntervalId: ReturnType<
+    typeof window.setInterval
+  > | null = null;
+  log: Logger;
 
-  constructor(options: Options) {
-    this.numPeers = options?.numPeersToUse ?? DEFAULT_NUM_PEERS_TO_USE;
+  private maintainPeersLock = false;
+
+  constructor(
+    protected core: BaseProtocol,
+    private connectionManager: ConnectionManager,
+    options: Options
+  ) {
+    this.log = new Logger(`sdk:${core.multicodec}`);
+    this.numPeersToUse = options?.numPeersToUse ?? DEFAULT_NUM_PEERS_TO_USE;
+    const maintainPeersInterval =
+      options?.maintainPeersInterval ?? DEFAULT_MAINTAIN_PEERS_INTERVAL;
+
+    /**
+     * Trigger maintainPeers when a new peer is identified.
+     * This is useful to make the process of finding and adding new peers more dynamic and efficient,
+     * instead of relying on the interval only.
+     */
+    this.core.addLibp2pEventListener("peer:identify", (evt) => {
```

Comment:
I am honestly a bit lost here, help please :) 
- when `"peer:identify"` fired - `maintainPeers` will be called;
- but then we anyway do this in `startMaintainPeersInterval` and even do this later in interval;

Why do we need to use event still?